File: rotatearray/ro.js
var stdin= process.openStdin();
stdin.addListener("data", function(d)
{
var x= d.toString().trim().split("\n");
let b= Number(x[0]);
let sub = JSON.parse(x[1]);
let k= Number(x[2]);

var res= rotate_array(sub,b,k);
let s='';
for(var i=0;i<b;i++)
{
s+=res[i]+" ";
}
console.log(s);
});



function rotate_array(arr,n,k)
{
	var k_step_rotation=[0];
    //Write your code here without removing the exising code.
	//the variable 'arr' contains array of integers of size n and integer k.
	//modified the array 'k_step_rotation' return the output of the program.
	return k_step_rotation;
    
}

File: rotatearray/ro.c

// Starter Code
#include <stdio.h>
//#include<stdio.h>
#include <stdlib.h>

// #include <stdio.h>
#include <string.h>
// Starter Code


int* rotate_array(int arr[],int n,int k)
{
    static int k_step_rotation[100000];
    //Write your code here without removing the exising code
	//the variable 'arr' contains array of integers of size n and integer k.
	//modified the array 'k_step_rotation' return the output of the program.
    return k_step_rotation;
}


//Backend Code
int main()
{
    int *arr;
    char str[100000];

    int n,k;
    scanf("%d", &n);
    int i = 0;

    arr = (int *)malloc(n * sizeof(int));
    int b[100];

    scanf("%s", str);
    char *token = strtok(str, "[");
    token = strtok(token, "]");
    token = strtok(token, ",");
    if (i < n)
    {
        while (token != NULL)
        {
            char *temp = token;
            int x = atoi(temp);
            *(arr + i) = x;
            token = strtok(NULL,",");
            i++;
        }
    }
    scanf("%d",&k);
    int *ans=rotate_array(arr,n,k);
    for (int i = 0; i<n; i++)
    {
        
        printf("%d ",*(ans+i));
    }

    return 0;
}

